Clarify which relay address to use for a LoadBalancer type ingress (#1380)
solo-io/gloo-mesh: docs/content/setup/prerequisites/enterprise_prerequisites.md

## Ingress Setup

> {{% notice note %}}This type of ingress is recommended if you are using GKE or Amazon EKS.{{% /notice %}}

Reviewer: I don't think this notice belongs here at the top level of this section. I'm also hesitant to provide this prescription, since there's a lot to consider when a user determines their ingress approach.

approach **does not** work by default with Kind cluster deployments, but is a good option for getting started if you're
running your clusters via a managed service like Google Kubernetes Engine or Amazon's Elastic Kubernetes Service.

> Assuming your ingress service is running on a node port, you can get the address of this ingress for use during [cluster registration]({{% versioned_link_path fromRoot="/setup/cluster_registration/enterprise_cluster_registration" %}}), run:

Reviewer: These instructions are for getting the relay server address for a LoadBalancer istio ingress, rather than a LoadBalancer enterprise-networking service

Reviewer: perhaps we should move this block below to the Istio Ingress section?

Author: Modified the beginning part a bit to make it more clear.

              number: 9900
```

You can set your ingress service as either a NodePort or LoadBalancer type.
In either case, you can get the address of this ingress for use
during [cluster registration]({{% versioned_link_path fromRoot="/setup/cluster_registration/enterprise_cluster_registration" %}}) by running:

{{< tabs >}}
{{< tab name="NodePort" codelang="yaml">}}

> MGMT_INGRESS_ADDRESS=$(kubectl get node -ojson | jq -r ".items[0].status.addresses[0].address")

Reviewer 2: This will only work if there is 1 node, we need to change this command to get the node on which the `istio-ingressgateway` is running

              number: 9900
```

Assuming your ingress service is running on a node port, you can get the address of this ingress for use
during [cluster registration]({{% versioned_link_path fromRoot="/setup/cluster_registration/enterprise_cluster_registration" %}}. Run:

```shell
MGMT_INGRESS_ADDRESS=$(kubectl get node -ojson | jq -r ".items[0].status.addresses[0].address")
MGMT_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
RELAY_ADDRESS=${MGMT_INGRESS_ADDRESS}:${MGMT_INGRESS_PORT}
```

> Another option is to set your ingress service as a LoadBalancer type.

Reviewer: I don't think we need to include this here, users can determine their ingress service type when they install istio.

              number: 9900
```

> Assuming your ingress service is running on a node port, you can get the address of this ingress for use

Reviewer: Let's update this to use our tabbed content view; we can have a tab for NodePort ingress services and another for LoadBalancer ingress services.

              number: 9900
```

You can set your ingress service as either a NodePort or LoadBalancer type. The

> LoadBalancer type is usually better, but sometimes your environment doesn't allow it.

Reviewer: nit: let's avoid being prescriptive here. A lot of our docs are centered around the kind-mgmt-cluster/kind-remote-cluster example, and it uses nodeport. I think we can remove this sentence.

> In either case, you can get the address of this ingress for use

Reviewer: also, the steps below assume that users named their ingress port "https". we should state that assumption

during [cluster registration]({{% versioned_link_path fromRoot="/setup/cluster_registration/enterprise_cluster_registration" %}}) by running:

{{< tabs >}}

> {{< tab name="NodePort" codelang="yaml">}}

Reviewer: ```suggestion
{{< tab name="NodePort" codelang="shell">}}
```

MGMT_INGRESS_ADDRESS=$(kubectl get node -ojson | jq -r ".items[0].status.addresses[0].address")
MGMT_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
RELAY_ADDRESS=${MGMT_INGRESS_ADDRESS}:${MGMT_INGRESS_PORT}
{{< /tab >}}

> {{< tab name="LoadBalancer" codelang="yaml">}}

Reviewer: ```suggestion
{{< tab name="LoadBalancer" codelang="shell">}}
```

MGMT_INGRESS_ADDRESS=$(kubectl  -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
MGMT_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
RELAY_ADDRESS=${MGMT_INGRESS_ADDRESS}:${MGMT_INGRESS_PORT}
{{< /tab >}}
{{< /tabs >}}

If you are using a NodePort and are failing to connect to relay,

> that means you have multiple nodes, and the istio-ingressgateway isn't

Reviewer: ```suggestion
that could mean you have multiple nodes, and the istio-ingressgateway isn't
```

From here, you need to find the address of this ingress to use
during [cluster registration]({{% versioned_link_path fromRoot="/setup/cluster_registration/enterprise_cluster_registration" %}}).

Some useful scripts for finding the ingress address can be found below. Note
that these scripts assume that 1) there is only one node, and 2) the ingress port is named "https":

{{< tabs >}}
{{< tab name="NodePort" codelang="shell">}}
MGMT_INGRESS_ADDRESS=$(kubectl get node -ojson | jq -r ".items[0].status.addresses[0].address")
MGMT_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
RELAY_ADDRESS=${MGMT_INGRESS_ADDRESS}:${MGMT_INGRESS_PORT}
{{< /tab >}}
{{< tab name="LoadBalancer" codelang="shell">}}
MGMT_INGRESS_ADDRESS=$(kubectl  -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
MGMT_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
RELAY_ADDRESS=${MGMT_INGRESS_ADDRESS}:${MGMT_INGRESS_PORT}
{{< /tab >}}
{{< /tabs >}}

> If you are using a NodePort and are failing to connect to relay,

Reviewer 3: Can we also highlight this assumption with a comment in the NodePort tab between L98 and L99?

              number: 9900
```

You can set your ingress service as either a NodePort or LoadBalancer type.
From here, you need to find the address of this ingress to use
during [cluster registration]({{% versioned_link_path fromRoot="/setup/cluster_registration/enterprise_cluster_registration" %}}).

> Some useful scripts for finding the ingress address can be found below (we assume that the ingress port is named https.):

Reviewer 4: ```suggestion
Some useful scripts for finding the ingress address can be found below (we assume that the ingress port is named https):
```

{{< tabs >}}
{{< tab name="NodePort" codelang="shell">}}
# Note that this command fetches the IP address of the first node.
# If you have multiple nodes, make sure to fetch the IP of the node

> # on which the istio-ingressgateway pod is deployed on (indicated by

Reviewer 4: ```suggestion
# on which the istio-ingressgateway pod is deployed (indicated by
```

> // the pod `spec.nodeName` field)

Reviewer 4: for shell commands should we use `#` for comments?
